include/cutlass/cutlass.h

```diff
@@ -115,205 +115,254 @@ static const int NumThreadsPerQuadPair = NumThreadsPerQuad * 2;
 #endif
 }
 
 // syncthreads
 
 CUTLASS_DEVICE void syncthreads() {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+    __syncthreads();
+#elif defined(__SYCL_Device_ONLY__)
   syclcompat::wg_barrier();
-#else
-  __syncthreads();
 #endif
 }
 
 CUTLASS_DEVICE int syncthreads_and(int cond) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+  return __syncthreads_and(cond);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
 #else
-  return __syncthreads_and(cond);
+  return 0;
 #endif
 }
 
 CUTLASS_DEVICE void syncwarp() {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+    __syncwarp();
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
-#else
-  __syncwarp();
 #endif
 }
 
 CUTLASS_DEVICE void threadfence() {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+    __threadfence();
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
-#else
-  __threadfence();
 #endif
 }
 
 // byte perm
 
 CUTLASS_DEVICE
 uint byte_perm(uint x, uint y, uint s) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+   return __byte_perm(x, y, s);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
+  return 0;
 #else
-  return __byte_perm(x, y, s);
+ return 0;
 #endif
 }
 
 // shfl
 
 CUTLASS_DEVICE
 uint shfl_up_sync(const unsigned mask, const uint var, const int delta, const int width = NumThreadsPerWarp) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+  return __shfl_up_sync(mask, var, delta, width);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
+  return 0;
 #else
-  return __shfl_up_sync(mask, var, delta, width);
+  return 0;
 #endif
 }
 
 CUTLASS_DEVICE
 uint shfl_down_sync(const unsigned mask, const uint var, const int delta, const int width = NumThreadsPerWarp) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+  return __shfl_down_sync(mask, var, delta, width);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
+  return 0;
 #else
-  return __shfl_down_sync(mask, var, delta, width);
+  return 0;
 #endif
 }
 
 CUTLASS_DEVICE
 uint shfl_sync(const unsigned mask, const uint var, const int delta, const int width = NumThreadsPerWarp) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__) 
+  return __shfl_sync(mask, var, delta, width);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
   assert(false);
+  return 0;
 #else
-  return __shfl_sync(mask, var, delta, width);
+  return 0;
 #endif
 }
 
 // math
 
 template <typename T>
 CUTLASS_DEVICE T hfma2(const T a, const T b, const T c) {
-#if defined(CUTLASS_ENABLE_SYCL)
+#if defined(__CUDA_ARCH__)
+  return hfma2(a, b, c);
+#elif defined(__SYCL_Device_ONLY__)
   // TODO: Add SYCL equivalent function
```

Comment:
@aacostadiaz , was there a reason `sycl::fma` was not used here ?
Acc. to the spec, It accepts sycl::vec of types float, double and half as well

PS - I am assuming that "Add SYCL equivalent function" means that there is no way to support this via sycl at the moment

Reply:
We can use `sycl::fma` here. The TODO comment means let's figure out its equivalent later when we can properly test it.